src/clustering/administration/auth/permissions.cc

```diff
@@ -145,7 +145,7 @@ void permissions_t::set_config(tribool config) {
 }
 
 void permissions_t::set_connect(tribool connect) {
-    guarantee(m_connect == make_optional(tribool::True));
+    guarantee(m_connect.has_value());
```

Comment:
For future reference, the bug was introduced in https://github.com/rethinkdb/rethinkdb/commit/e705583d8c6dfd87ac90cbd9946b1810fee13552#diff-5f4c6e389529139aa0cef0416cf8bcf4R149 which changed the behavior from `m_connect != null` to `m_connect == true`.